reflectapi/src/codegen/openapi.rs

```diff
@@ -0,0 +1,595 @@
+                InlineOrRef::Inline(schema) => schema,
+                InlineOrRef::Ref(r) => {
+                    assert!(
+                        self.components.schemas.remove(&name).is_some(),
+                        "remove stub"
+                    );
+                    return InlineOrRef::Ref(r);
+                }
+            },
+            crate::Type::Enum(adt) => {
+                if adt.name == "std::option::Option" {
+                    // Special case `Option` to generate a nicer spec.
+                    Schema::OneOf {
+                        subschemas: vec![
+                            InlineOrRef::Inline(Schema::Flat(FlatSchema {
+                                description: "Null".to_owned(),
+                                ty: Type::Null,
+                            })),
+                            self.convert_type_ref(schema, kind, ty_ref.arguments().next().unwrap()),
+                        ],
+                        discriminator: None,
+                    }
+                } else {
+                    match self.enum_to_schema(
+                        schema,
+                        kind,
+                        &adt.clone().instantiate(&ty_ref.arguments),
+                    ) {
+                        InlineOrRef::Inline(schema) => schema,
+                        InlineOrRef::Ref(r) => {
+                            assert!(
+                                self.components.schemas.remove(&name).is_some(),
+                                "remove stub"
+                            );
+                            return InlineOrRef::Ref(r);
+                        }
+                    }
+                }
+            }
+            crate::Type::Primitive(prim) => {
+                assert_eq!(ty_ref.arguments.len(), prim.parameters.len());
+                let ty = match prim.name() {
+                    "f32" | "f64" => Type::Number,
+                    "i8" | "i16" | "i32" | "i64" | "i128" | "isize" | "u8" | "u16" | "u32"
+                    | "u64" | "u128" | "usize" => Type::Integer,
+                    "bool" => Type::Boolean,
+                    "char" | "std::string::String" => Type::String,
+                    "std::marker::PhantomData" | "unit" => Type::Null,
+                    "std::vec::Vec" | "std::array::Array" | "std::collections::HashSet" => {
+                        Type::Array {
+                            items: Box::new(self.convert_type_ref(
+                                schema,
+                                kind,
+                                ty_ref.arguments().next().unwrap(),
+                            )),
+                            unique_items: prim.name() == "std::collections::HashSet",
+                        }
+                    }
+                    // Treat as transparent wrappers? Not sure why these types are relevant to an API anyway @Andrey?
+                    "std::boxed::Box" | "std::cell::Cell" | "std::cell::RefCell"
```

Comment:
Can you have a look at this @avkonst? My opinion is that these pointer/wrapper types `Wrapper<T>` should be represented in the API the same as `T`. It seems irrelevant to the API how they are represented in memory.

Reply:
for openapi and typescript codegen, use fallback type whenever it is available on the type definition. These pointers fallback to T. Have a look how TS codegen resolves these. Same applies to openapi.
The reason we have to keep these explicitly defined (unless a user tells to transform to fallback explictly with an attribute), is because Rust recursive references would not work for the rust codegen. For example Struct of T with field of type T is not possible.